src/app/extensions/seo/store/seo/seo.effects.ts

```diff
@@ -17,12 +20,20 @@ import { SeoActionTypes, SetSeoAttributes } from './seo.actions';
 
 @Injectable()
 export class SeoEffects {
+  cannonicalLink: HTMLLinkElement;
+
   constructor(
     private actions$: Actions,
     private store: Store<{}>,
     private meta: MetaService,
-    private translate: TranslateService
-  ) {}
+    private translate: TranslateService,
+    @Inject(DOCUMENT) private doc: Document
+  ) {
+    this.cannonicalLink = this.doc.createElement('link');
```

Comment:
should fix the typo "cannonical" => "canonical".
No cannons 💣 😎

Reply:
You should consider also checking if there is an existing canonical link set (to his may appear when changing a route). In that case you shouldn't just add the link but update the existing one.
I've implemented in a similar way here: https://github.com/d-koppenhagen/d-koppenhagen.de/commit/1fd231a061803dffdb7bae700090e0f18da14cb7

Reply:
Implementing it here would be even better: https://github.com/fulls1z3/ngx-meta/issues/44

Reply:
ngx-meta dont support canonical-links so i will do in our seo.effects

Reply:
That was my suggestion, to implement it there... 😅